### PR TITLE
Epistemic Scoring for monaibundle app

### PR DESCRIPTION
**Pull request by tangy5**
Refactored the PR for Epistemic scoring in monaibundle app. 

Due to the latest bigs changes on **bundleInferTask"", and logics for initializing/loading monaibundle app. 
The epistemic scoring function is refactored to adapt latest changes. 

Verified Sample run:

```bash
# Manual define epistemic scoring parameters
monailabel start_server \
  --app workspace/monaibundle \
  --studies workspace/images \
  --conf models spleen_ct_segmentation_v0.2.0,swin_unetr_btcv_segmentation_v0.2.0 \
  --conf epistemic_model spleen_ct_segmentation_v0.2.0
  --conf epistemic_max_samples 0 \
  --conf epistemic_simulation_size 5
```

Small changes to get bundle models, users can set auth_token for github api rate limits.

### CODE REVIEW

File: sample-apps/monaibundle/main.py [hunk 1]
```diff
@@ -30,6 +34,15 @@
...
         self.models = get_bundle_models(app_dir, conf)
+        # Add Epistemic model for scoring
+        self.epistemic_models = (
+            get_bundle_models(app_dir, conf, conf_key="epistemic_model") if conf.get("epistemic_model") else None
+        )
+        if self.epistemic_models:
+            # Get epistemic parameters
+            self.epistemic_max_samples = int(conf.get("epistemic_max_samples", "0"))
+            self.epistemic_simulation_size = int(conf.get("epistemic_simulation_size", "5"))
+            self.scoring_dropout = float(conf.get("scoring_dropout", "0.2"))
```
Comment (SachidanandAlle) [inline sample-apps/monaibundle/main.py:45]: let this also start as `epistemic_dropout` 
And we need to mention these as part of readme.. something like this..
https://github.com/Project-MONAI/MONAILabel/tree/main/sample-apps/monaibundle#additional-configs

Comment (tangy5) [reply sample-apps/monaibundle/main.py]: Thanks for suggestions! changed to epistemic_dropout. Added configs to readme.

File: sample-apps/monaibundle/README.md [hunk 2]
```diff
@@ -72,14 +78,38 @@ monailabel start_server --app workspace/monaibundle --studies workspace/images -
...
+|---------------------------|-----------------|---------------------------------------------------------------------------------------------|
+| zoo_source                | string          | _Default value:_ github                                                                     |
+| zoo_repo                  | string          | _Default value:_ Project-MONAI/model-zoo/hosting_storage_v1                                 |
+| preload                   | true, **false** | Preload model into GPU                                                                      |
+| skip_trainers             | true, **false** | Skip adding training tasks (Run in Infer mode only)                                         |
+| epistemic_max_samples     | int             | _Default value:_ 0    ;  Epistemic scoring parameters                                       |
+| epistemic_simulation_size | int             | _Default value:_ 5    ;  Epistemic simulation size parameters                               |
+| epistemic_dropot          | float           | _Default value:_ 0.2  ;  Epistemic scoring parameters: Dropot rate for scoring models       |                               |
```
Comment (SachidanandAlle) [inline sample-apps/monaibundle/README.md:115]: epistemic_dropot => epistemic_dropout

File: sample-apps/monaibundle/main.py [hunk 3]
```diff
@@ -30,6 +34,15 @@
...
         self.models = get_bundle_models(app_dir, conf)
+        # Add Epistemic model for scoring
+        self.epistemic_models = (
+            get_bundle_models(app_dir, conf, conf_key="epistemic_model") if conf.get("epistemic_model") else None
+        )
+        if self.epistemic_models:
+            # Get epistemic parameters
+            self.epistemic_max_samples = int(conf.get("epistemic_max_samples", "0"))
+            self.epistemic_simulation_size = int(conf.get("epistemic_simulation_size", "5"))
+            self.epistemic_dropot = float(conf.get("epistemic_dropot", "0.2"))
```
Comment (SachidanandAlle) [inline sample-apps/monaibundle/main.py:45]: same spelling error

File: sample-apps/monaibundle/main.py [hunk 4]
```diff
@@ -74,6 +87,32 @@ def init_strategies(self) -> Dict[str, Strategy]:
...
+            return methods
+
+        for n, b in self.epistemic_models.items():
+            # Create BundleInferTask task with dropout instantiation for scoring inference
+            i = BundleInferTask(
+                b,
+                self.conf,
+                train_mode=True,
+                skip_writer=True,
+                dropout=self.epistemic_dropot,
```
Comment (SachidanandAlle) [inline sample-apps/monaibundle/main.py:102]: here..

File: sample-apps/monaibundle/README.md (hunk 2 above)
Comment (tangy5) [reply sample-apps/monaibundle/README.md]: Good Catch! Fixed, thanks.

**Review by SachidanandAlle — APPROVED** [review]

**Pull request merged**